cli/commands/project_create.go

```diff
@@ -0,0 +1,71 @@
+package commands
+
+import (
+	"fmt"
+	"os"
+	"os/exec"
+	"strings"
+
+	"github.com/phase2/rig/cli/util"
+	"github.com/urfave/cli"
+)
+
+type ProjectCreate struct {
+	BaseCommand
+}
+
+func (cmd *ProjectCreate) Commands() []cli.Command {
+	create := cli.Command{
+		Name:        "create",
+		Aliases:     []string{},
+		Usage:       "Run a yeoman generator in a container to create project configuration files",
+		ArgsUsage:   "[optional type] [optional args]",
+		Description: "The type is the generator to run with args passed to that generator. If using flag arguments use -- before specifying type and arguments.",
+		Flags: []cli.Flag{
+			cli.StringFlag{
+				Name:  "image",
+				Usage: "Docker image to use if default outrigger/generator is not desired",
+			},
+		},
+		Before:      cmd.Before,
+		Action:      cmd.Create,
+	}
+
+	return []cli.Command{create}
+}
+
+// Run a docker image to execute the desired generator
+func (cmd *ProjectCreate) Create(ctx *cli.Context) error {
+	image := ctx.String("image")
+	if (image == "") {
+		image = "outrigger/generator"
+	}
+
+	cwd, err := os.Getwd();
+	if (err != nil) {
+		cmd.out.Error.Printf("Couldn't determine current working directory: %s", err)
+		os.Exit(1)
+	}
+
+	cmd.out.Verbose.Printf("Executing container %s with arguments %s", image, strings.Join(ctx.Args(), " "))
+
+	// keep passed in args as distinct elements or they will be treated as
+	// a single argument containing spaces when the container gets them
+	args := []string{
+		"run",
+		"--rm",
+		"-it",
+		"-v", fmt.Sprintf("%s:/generated", cwd),
+		image,
+		"yo", "--no-insight",
```

Comment:
Yeah, you don't even need the "yo" command.  Just pass any specified "type" and any additional specified "args".  The container will run yo directly and will add the --skip-install and --no-insight options.

Basically, rig doesn't have to know anything about this being "yeoman".